packages/blockchain-wallet-v4-frontend/src/modals/TransactionReport/selectors.js

```diff
@@ -87,18 +96,39 @@ export const getBchData = createSelector(
           d.value_then,
           d.value_now,
           d.exchange_rate_then,
-          d.tx
+          d.tx,
+          d.note
         ],
         data
       )
       return [headers].concat(transformedData)
     }
     const start = prop('start', formValues)
     const end = prop('end', formValues)
+    const notes = notesR.getOrElse({})
     return {
-      csvData: dataR.map(transform).getOrElse(undefined),
+      csvData: dataR
+        .map(assocBCHNotes(notes))
+        .map(transform)
+        .getOrElse(undefined),
       isValidStartDate: date => isValidBchStartDate(date, end),
       isValidEndDate: date => isValidBchEndDate(date, start)
     }
   }
 )
+
+const assocBTCNotes = curry((wallet, transactions) => {
+  return transactions.map(transaction => {
+    const hash = prop('tx', transaction)
+    const note = TXNotes.selectNote(hash, Wallet.selectTxNotes(wallet))
```

Comment:
`Wallet.selectTxNotes(wallet)` could be made a selector, and used instead of selecting a full wallet